pkg/i2gw/providers/ingressnginx/canary.go

```diff
@@ -47,7 +47,7 @@ func canaryFeature(ingressResources i2gw.InputResources, gatewayResources *i2gw.
 			key := types.NamespacedName{Namespace: path.ingress.Namespace, Name: common.RouteName(rg.Name, rg.Host)}
 			httpRoute, ok := gatewayResources.HTTPRoutes[key]
 			if !ok {
-				panic("HTTPRoute not exists - this should never happen")
+				return field.ErrorList{field.InternalError(nil, fmt.Errorf("HTTPRoute does not exist"))}
```

Comment:
On lines 44-45 we get errors from `calculateBackendRefWeight`. Although you replaced panic with returning an error, the errors above are still invisible and the user can not know whats wrong.

The concern is relevant for changes in Kong as well.

Reply:
That's a good catch. In those cases, instead of panicking or returning uninformative errors such as "HTTPRoute does not exist," we could `continue` to the next iteration. All the informative errors will be collected and returned at the end of the for-loop

WDYT @LiorLieberman @mlavacca?

Reply:
`continue` plus a comment on top of this to explain sounds good to me. 
Also since you changed `kong/converter.go` to the logic [proposed](https://github.com/kubernetes-sigs/ingress2gateway/pull/152#discussion_r1558876461), I think is ok to log and return an error because in kong this indeed should never happen.

Reply:
> continue plus a comment on top of this to explain sounds good to me.

Done

> Also since you changed kong/converter.go to the logic https://github.com/kubernetes-sigs/ingress2gateway/pull/152#discussion_r1558876461, I think is ok to log and return an error because in kong this indeed should never happen.

I'm not exactly sure what you mean here

Reply:
for all kong featureParsers changes you have added, (returning error instead of panic) maybe we should add that this should never happen? or something more clear to the user?
Because this indeed should never happen now with the change you added to check the errors and return before calling the featureParsers in `kong/converter.go`

I hope this is a bit more clear

Reply:
OK got it now. I added back the "this should never happen" message. If we indeed encounter this error at some point for whatever reason we'll figure out how to address it better